CMS-Enterprise/mint-app
# [MINT-3220] Add milestone to matrix modal

# MINT-3220

## Description


- Added modal/form to add common milestone to MTO
- Adjusted `<Multiselect />` to allow for grouped options and relative styling
- Added unit tests

> [!NOTE]
> This PR also set up the query and form work to support custom solutions.  This work is not yet in place for the BE, so it's commented with a `TODO` tag in the file

## How to test this change

1.  Navigate to MTO matrix milestone library
2. Click Add to matrix on any card
3. Select from a list of the solutions or select none
4. Verify the solutions are not duplicated if they are in the common list
5. Verify Figma/success/error

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
